spring-cloud-generator/src/main/java/com/google/api/generator/spring/composer/SpringAutoConfigClassComposer.java

```diff
@@ -0,0 +1,1146 @@
+import com.google.api.generator.gapic.utils.JavaStyle;
+import com.google.api.generator.spring.composer.comment.SpringAutoconfigCommentComposer;
+import com.google.api.generator.spring.utils.ComposerUtils;
+import com.google.api.generator.spring.utils.LoggerUtils;
+import com.google.api.generator.spring.utils.Utils;
+import com.google.cloud.spring.autoconfigure.core.GcpContextAutoConfiguration;
+import com.google.cloud.spring.core.Credentials;
+import com.google.cloud.spring.core.DefaultCredentialsProvider;
+import com.google.common.base.CaseFormat;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.AutoConfiguration;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+
+public class SpringAutoConfigClassComposer implements ClassComposer {
+  private static final SpringAutoConfigClassComposer INSTANCE = new SpringAutoConfigClassComposer();
+
+  private static final Map<String, TypeNode> STATIC_TYPES = createStaticTypes();
+  private static final Statement EMPTY_LINE_STATEMENT = EmptyLineStatement.create();
+
+  private SpringAutoConfigClassComposer() {}
+
+  public static SpringAutoConfigClassComposer instance() {
+    return INSTANCE;
+  }
+
+  @Override
+  public GapicClass generate(GapicContext context, Service service) {
+    String packageName = Utils.getSpringPackageName(service.pakkage());
+    Map<String, TypeNode> dynamicTypes = createDynamicTypes(service, packageName);
+    String serviceName = service.name();
+    String serviceNameLowerCamel = JavaStyle.toLowerCamelCase(serviceName);
+    String serviceNameLowerHyphen = CaseFormat.UPPER_CAMEL.to(CaseFormat.LOWER_HYPHEN, serviceName);
+    String className = Utils.getServiceAutoConfigurationClassName(service);
+    String credentialsProviderName = serviceNameLowerCamel + "Credentials";
+    String transportChannelProviderName = "default" + serviceName + "TransportChannelProvider";
+    String clientName = serviceNameLowerCamel + "Client";
+    GapicClass.Kind kind = Kind.MAIN;
+
+    GapicServiceConfig gapicServiceConfig = context.serviceConfig();
+
+    Expr thisExpr = ValueExpr.withValue(ThisObjectValue.withType(dynamicTypes.get(className)));
+    Transport transport = context.transport();
+    // TODO(emmwang): this condition is adapted from latest gapic-generator-java changes as part of
```

Comment:
Can we fix this TODO as part of this PR?

Reply:
Thanks for catching this! Will update as part of this PR.